ibis/expr/timecontext.py

```diff
@@ -127,6 +127,40 @@ def canonicalize_context(
     return begin, end
 
 
+def construct_multi_index_series(
+    series: pd.Series, frame: pd.DataFrame
+) -> pd.Series:
+    """ Construct a pd.MultiIndex of IntIndex and 'time'
+
+    In window execution, the result Series of udf may need
+    to be trimmed by timecontext. In order to do so, 'time'
+    must be added as an index to the Series. We extract
+    time column from the parent Dataframe `frame`.
+    See `trim_with_timecontext` in execution/window.py for
+    trimming implementation.
+
+    Parameters
+    ----------
+    series: pd.Series
+    frame: pd.DataFrame
+
+    Returns
+    -------
+    pd.Series
+    """
+    if TIME_COL == frame.index.name:
+        time_index = frame.index
+    elif TIME_COL in frame:
+        time_index = frame.set_index(TIME_COL).index
+    else:
+        raise com.IbisError(f'"time" column not present in DataFrame {frame}')
```

Comment:
When do we raise this error? We do not always require a time column, right?

Reply:
You are right, this error raises when 'time' column is not present. Note that any reference to this function there is a if statement, only when there is `timecontext` defined will we call this `construct_time_context_aware_series`.

Reply:
Can we assert when entering the function that time column exists?

Reply:
Yes, it's just the same as `assert TIME_COL in frame or TIME_COL == frame.index.name` when entering the function, it's just because I want to branch for these two cases again to extract time_index so I put this as a raise error in else branch.

Reply:
Oh ok